lib/array_equals.rb

```diff
@@ -1,5 +1,24 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
+
 def array_equals(array1, array2)
-  raise NotImplementedError
+  i = 0
+  if (array1 == nil) && (array2 == nil)
+    return true
+  elsif (array1 == nil) && (array2 != nil)
```

Comment:
When the body of an if statement is a return, then the else is kind of redundant. It's not incorrect, and I think the elseif improves readability over a bare if. Just something to think about.

Also, you could collapse this condition with the next one into (array1 == nil || array2 == nil).